File: sum_class.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import rospy
from yh_sum.msg import yh_sum_msg

class MyClass:
    def __int__(self):
        self.sub = rospy.SubscribeListener("yh_sum_topic",yh_sum_msg, self.msgCallback)
        self.old_data = 0
    
    def msgCallback(self,msg):
            
        print(self.old_data + msg.data)
        self.old_data = msg.data
        

def listner():
    rospy.init_node("yh_sum_sub", anonymous=True)
    my_class = MyClass()    
    rospy.spin()

if __name__ == "__main__":
    listner()

File: sum_class.cpp
#include "ros/ros.h"
#include "yh_sum/yh_sum_msg.h"

int old_data = 0;
class MyClass
{
    public:
        MyClass()
        {
            sub = nh.subscriber("yh_sum_topic",100,&MyClass::msgCallback,this);
        }
    void msgCallback(const yh_sum::yh_sum_msg::ConstPtr& msg)
    {
        printf("%d\n", old_data + msg->data);
        old_data = msg->data;

    }
    private:
        ros::NodeHandle nh;
        ros::Subscriber sub;
        int old_data = 0;

}



int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_sum_sub");
    MyClass my_class;
    ros::spin();  //프로그램 종료 방지 대기

    return 0;

}
